File: JisooPark/Week010/1128_TestBank/pr77484.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    
    map<int, int> nums; 
    for(int n : win_nums)
    {
        nums[n] = 0;
    }
    
    int zeroCount = 0;
    
    for(int n : lottos)
    {
        if(n == 0)
        {
            zeroCount++;
            continue;
        }
        
        if(nums.find(n) != nums.end())
        {
            nums[n]++;
        }
    }
    
    int winCount = 0;
    for(auto n : nums)
    {
        winCount += n.second;       
    }

    answer.push_back(min(6 - (winCount + zeroCount) + 1, 6));
    answer.push_back(min(6 - winCount + 1, 6));
    
    return answer;
}

File: HanByeolKo/week001_010/week010/day1128_TestBank/Pr77484.java
package week001_010.week010.day1128_TestBank;

public class Pr77484 {
    public int[] solution(int[] lottos, int[] win_nums) {
        int match = 0;
        int zero = 0;

        for (int lotto : lottos) {
            if (lotto == 0) {
                zero++;
                continue;
            }
            for (int win : win_nums) {
                if (lotto == win) {
                    match++;
                }
            }
        }

        int max = calculate(match + zero);
        int min = calculate(match);

        int[] answer = {max, min};

        return answer;
    }

    public static int calculate(int count) {
        return switch (count) {
            case 2 -> 5;
            case 3 -> 4;
            case 4 -> 3;
            case 5 -> 2;
            case 6 -> 1;
            default -> 6;
        };
    }

}
